### DISCUSSION TITLE
How to handle "Skipping Draco compression on non-indexed primitive"

### DISCUSSION THREAD

**Discussion by marcofugaro** [donmccurdy/glTF-Transform#394] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**

Hi, I am using the JS api of the draco encoder in conjunction with the three.js' GLTFExporter like this:

```js
import { WebIO } from '@gltf-transform/core';
import { KHRONOS_EXTENSIONS, DracoMeshCompression } from '@gltf-transform/extensions';

const io = new WebIO()
  .registerExtensions(KHRONOS_EXTENSIONS)
  .registerDependencies({
    'draco3d.encoder': await new DracoEncoderModule(),
  });

new GLTFExporter().parse(scene, (gltf) => {
  const document = io.readJSON({
    json: gltf,
    resources: {}, 
  });

  document.createExtension(DracoMeshCompression)
    .setRequired(true)

  const { json: compressedGltf } = io.writeJSON(document);
});
```

Whenever I run this code, it logs this:

![image](https://user-images.githubusercontent.com/7217420/138495098-9f2d98dc-78e8-42c2-8712-58d3992c5ce2.png)

And the gltf is of course not compressed.

The input geometry looks like this, as you can see it's a non-indexed geometry.
![image](https://user-images.githubusercontent.com/7217420/138495416-69c10c00-1396-4043-b78f-261258ab3f3c.png)

I can provide an executable example if needed.

**Describe the solution you'd like**
Any way to make it work with non-indexed geometries?

**Additional context**
Related to this https://github.com/mrdoob/three.js/pull/22227#issuecomment-897310955

**Comment by donmccurdy** [comment, marked as answer]
The primitive needs to be indexed for compression, and that isn't done automatically because it's generally lossy... You can do this manually with the `weld` function though:

```js
import { weld } from '@gltf-transform/functions';

await document.transform(weld({tolerance: 0.0001}));

const glb = io.writeBinary(document);
```

***

```js
const { json: compressedGltf } = io.writeJSON(document);
```

Note that resources are returned here as well; if you need a single file it's best to write to a binary GLB.

**Reply by marcofugaro** [reply]
Thanks! It worked
